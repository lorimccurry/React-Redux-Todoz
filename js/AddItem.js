import React from 'react'
import { connect } from 'react-redux'
import { setItemInput } from './actionCreators'

class AddItem extends React.Component {
  constructor () {
    super()

    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleChange (e) {
    this.props.dispatch(setItemInput(e.target.value))
  }

  handleSubmit (e) {
    e.preventDefault()
    this.props.onSubmit()
    this.props.dispatch(setItemInput(''))
  }

  render () {
    return (
      <form onSubmit={this.handleSubmit}>
        <fieldset>
          <legend>Gimme tha todoz:</legend>
          <input
            type='text'
            placeholder='Get coffee'
            value={this.props.itemInput}
            onChange={this.handleChange}
            autoFocus
          />
        </fieldset>
      </form>
    )
  }
}

const { string, func } = React.PropTypes
AddItem.propTypes = {
  itemInput: string,
  onSubmit: func,
  dispatch: func
}

const mapStateToProps = (state) => {
  return {
    itemInput: state.itemInput
  }
}

export default connect(mapStateToProps)(AddItem)
