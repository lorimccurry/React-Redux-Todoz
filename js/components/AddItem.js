import React from 'react'
import { connect } from 'react-redux'
import { addItem } from '../actions/actionCreators'
import { FormGroup, FormControl, ControlLabel } from 'react-bootstrap'

class AddItem extends React.Component {
  constructor () {
    super()

    this.state = {
      itemInput: ''
    }

    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleChange (e) {
    this.setState({itemInput: e.target.value})
  }

  handleSubmit (e) {
    e.preventDefault()
    this.props.dispatchAddItem(this.state.itemInput)
    this.setState({itemInput: ''})
  }

  render () {
    return (
      <form onSubmit={this.handleSubmit}>
        <FormGroup>
          <ControlLabel>Gimme tha todoz:</ControlLabel>
          <FormControl
            type='text'
            placeholder='Get coffee'
            value={this.state.itemInput}
            onChange={this.handleChange}
            autoFocus
            bsSize='lg'
          />
        </FormGroup>
      </form>
    )
  }
}

const { func, string } = React.PropTypes
AddItem.propTypes = {
  dispatchAddItem: func,
  itemInput: string
}

const mapDispatchToProps = (dispatch) => {
  return {
    dispatchAddItem: (text) => {
      dispatch(addItem(text))
    }
  }
}

export const Unwrapped = AddItem

export default connect(null, mapDispatchToProps)(AddItem)
