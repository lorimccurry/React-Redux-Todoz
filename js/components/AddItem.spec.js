import React from 'react'
import { Provider } from 'react-redux'
import { shallow, mount, render } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { spy } from 'sinon'
import { Unwrapped as UnwrappedAddItem } from './AddItem'

describe('AddItem', () => {
  it('AddItem snapshot test', () => {
    const component = shallow(<UnwrappedAddItem />)
    const tree = shallowToJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('should render a form and an input', () => {
    const component = render(<UnwrappedAddItem />) 
    expect(component.find('form').length).toEqual(1)
    expect(component.find('input').length).toEqual(1)
  })

  it('should accept input', () => {
    const component = mount(<UnwrappedAddItem />)
    const input = component.find('input')
    input.simulate('change', {target: {value: 'get beer'}})
    expect(component.state('itemInput')).toEqual('get beer')
    expect(input.prop('value')).toEqual('get beer')
  })

  it('should call dispatchAddItem when form is submitted', () => {
    const addItemSpy = spy()
    const component = mount(<UnwrappedAddItem dispatchAddItem={addItemSpy} />)
    component.setState({itemInput: 'get beer'})
    const form = component.find('form')
    form.simulate('submit')

    expect(addItemSpy.calledOnce).toEqual(true)
    expect(addItemSpy.calledWith('get beer')).toEqual(true)
  })
})

