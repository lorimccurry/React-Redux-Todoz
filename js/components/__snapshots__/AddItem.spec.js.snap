// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItem AddItem snapshot test 1`] = `
<form
  onSubmit={[Function]}
>
  <FormGroup
    bsClass="form-group"
  >
    <ControlLabel
      bsClass="control-label"
      srOnly={false}
    >
      Gimme tha todoz:
    </ControlLabel>
    <FormControl
      autoFocus={true}
      bsClass="form-control"
      bsSize="lg"
      componentClass="input"
      onChange={[Function]}
      placeholder="Get coffee"
      type="text"
      value=""
    />
  </FormGroup>
</form>
`;
